### ISSUE TITLE
GameCube LAN does not work.

### ISSUE THREAD

**Issue by nikkiwritescode** [mholtkamp/retro-league#2]
I've been testing this in my home LAN but the systems simply sit on Searching... forever. I have verified on the router side that the systems are properly connected, so it seems to be a software issue of some sort.

I know you don't have a Broadband Adapter to be able to test this, but in exchange for getting this (and the future Online mode) working, I will be glad to purchase and ship a BBA to you to assist with this. 

Thanks for all your hard work!

**Comment by mholtkamp** [comment]
Thank you for testing out the GameCube LAN status! I can't guarantee that I'll be able to fix it though. I think I'll pick up a BBA for myself anyway, and at least I can take a quick a look to see if there's something obviously wrong. I'll keep you updated.

**Comment by Sir-LoLz** [comment]
I have a BBA so I would be happy to help. Though Dolphin is able to emulate the BBA as well (and even has cross-play with real hardware) so its compatibility is good. 

If you have Ghidra you may want to look at the symbol map for MKDD. it lists all the functions in the game and even the code that initiates/controls the BBA

**Comment by Sir-LoLz** [comment]
You may be happy to know that it does actually detect the BBA code in your app. It's just that data does not seem to be flowing through it.

**Comment by mholtkamp** [comment]
Hey! That is good news :) Hmm, is it possible to run multiple instances of Dolphin and attempt to connect to each other using the BBA emulation? That would be really handy. We could add some logging messages to track down what's happening.

**Comment by mholtkamp** [comment]
Also @nikkiwritescode , I just released the full source code. Not sure if you were interested in that, but I thought I should mention it.

**Comment by Sierron** [comment]
Yeah, it is possible to have multiple dolphins on one system. Dolphin has two plugins for that. Tap Adapters and the Xlink Kai Plugin.

We've used Tap Adapters from Openvpn before but they are unrealiable. Nowadays we use Xlink Kai (https://www.teamxlink.co.uk/) to get LAN games online. It grabs the traffic of these games and fowards it to other players.

We've tried to use that approach but couldn't find each other.

Yeah, logging sounds useful to see what's going on.

**Comment by nikkiwritescode** [comment]
> Also @nikkiwritescode , I just released the full source code. Not sure if you were interested in that, but I thought I should mention it.

I might have missed it somewhere, but could you post instructions somewhere for how to compile the GameCube .dol? I don't have devkitPro, so I'd probably need to do some workstation setup to even get to the point where compiling for GCN is possible. The first step for resolving this, I think, would be adding on-screen logging as you mentioned. I'm a little unsure if the BBA is even getting initialized at this point. Fingers crossed that it's a fairly simple change and not a more comprehensive incompatibility! I seem to remember the network adapters for GCN being fairly standard, though.

**Comment by teamxlink-dev** [comment]
Hey everyone, was this issue with LAN solved? I'd love to give this a shot over Kai 👍

**Comment by Papy-G** [comment]
Hey there! Now we have the ENCJ60 support trhough SWISS, is there a way it can be supported?
The SWISS team has achieved BBA emulation through that device, their code is available to help if needed.
